pennylane/transforms/defer_measurements.py

```diff
@@ -179,11 +190,15 @@ def func(x, y):
         else:
             new_operations.append(op)
 
-    new_tape = QuantumTape(
-        new_operations[: -len(tape.measurements)],
-        new_operations[-len(tape.measurements) :],
-        shots=tape.shots,
-    )
+    new_measurements = []
+
+    for mp in tape.measurements:
+        if mp.mv is not None:
+            wire_map = {m.wires[0]: control_wires[m.id] for m in mp.mv.measurements}
+            mp = qml.map_wires(mp, wire_map=wire_map)
```

Comment:
will this map the wires for the measurements? should it?

Reply:
Yes. I updated `MeasurementProcess.map_wires` to also map wires for the mid-circuit measurements that `mp.mv` stores.